Use consistent terms regarding on-budget and off-budget accounts (#504)
actualbudget/docs: docs/accounts/index.md

## Off-budget accounts

Actual makes a distinction between accounts being **on**-budget or **off**-budget.

**Off-budget** accounts don't effect the budget and are meant to track stuff like investments and mortgages. Transactions in off-budget accounts can't be categorized; they simply track balances over time.

**On-budget** accounts affect the budget, and transactions can be categorized. These are accounts where you want to track cash flow and use the budget, like checking accounts and credit cards.

> Depending on your usage, savings accounts can either be on or off the budget. If you're not sure, we recommend keeping it on-budget at the start.

Reviewer: ```suggestion
Depending on your usage, savings accounts can be on or off the budget. If you're not sure, we recommend keeping them on budget at the start.
```
